src/arviz_stats/psens.py

```diff
@@ -0,0 +1,213 @@
+        - "strong prior / weak likelihood" if the prior sensitivity is above threshold
+        and the likelihood sensitivity is below the threshold
+        - "-" otherwise
+    """
+    pssdp = psens(data, group="log_prior")["posterior"].to_dataset()
+    pssdl = psens(data, group="log_likelihood")["posterior"].to_dataset()
+
+    joined = xr.concat([pssdp, pssdl], dim="component").assign_coords(
+        component=["prior", "likelihood"]
+    )
+
+    n_vars = np.sum([joined[var].size // 2 for var in joined.data_vars])
+
+    psens_df = pd.DataFrame(
+        (np.full((cast(int, n_vars), 2), np.nan)), columns=["prior", "likelihood"]
+    )
+
+    indices = []
+    for i, (var_name, sel, isel, values) in enumerate(
+        xarray_var_iter(joined, skip_dims={"component"})
+    ):
+        psens_df.iloc[i] = values
+        indices.append(labeller.make_label_flat(var_name, sel, isel))
+    psens_df.index = indices
+
+    def _diagnose(row):
+        if row["prior"] >= threshold and row["likelihood"] >= threshold:
+            return "prior-data conflict"
+        if row["prior"] > threshold > row["likelihood"]:
+            return "strong prior / weak likelihood"
+
+        return "-"
+
+    psens_df["diagnosis"] = psens_df.apply(_diagnose, axis=1)
+
+    return psens_df.round(round_to)
+
+
+# get_log_likelihood and get_log_prior functions should be somewhere else
+def get_log_likelihood(idata, var_name=None, single_var=True):
+    """Retrieve the log likelihood dataarray of a given variable."""
+    if (
+        not hasattr(idata, "log_likelihood")
+        and hasattr(idata, "sample_stats")
+        and hasattr(idata.sample_stats, "log_likelihood")
+    ):
+        warnings.warn(
+            "Storing the log_likelihood in sample_stats groups has been deprecated",
+            DeprecationWarning,
+        )
+        return idata.sample_stats.log_likelihood
+    if not hasattr(idata, "log_likelihood"):
+        raise TypeError("log likelihood not found in inference data object")
+    if var_name is None:
+        var_names = list(idata.log_likelihood.data_vars)
+        if len(var_names) > 1:
+            if single_var:
+                raise TypeError(
+                    f"Found several log likelihood arrays {var_names}, var_name cannot be None"
+                )
+            return idata.log_likelihood[var_names]
+        return idata.log_likelihood[var_names[0]]
+
+    try:
+        log_likelihood = idata.log_likelihood[var_name]
+    except KeyError as err:
+        raise TypeError(f"No log likelihood data named {var_name} found") from err
+    return log_likelihood
+
+
+def get_log_prior(idata, var_names=None):
+    """Retrieve the log prior dataarray of a given variable."""
+    if not hasattr(idata, "log_prior"):
+        raise TypeError("log prior not found in inference data object")
+    if var_names is None:
+        var_names = list(idata.log_prior.data_vars)
+    return idata.log_prior.to_dataset()[var_names]
```

Comment:
```suggestion
# somewhere in the imports: from collections.abc import Hashable
    if var_names is None:
        return idata.log_prior
    if isinstance(var_names, Hashable):
        return idata.log_prior[[var_names]]
    return idata.log_prior.to_dataset()[var_names]
```

I think this will guarantee dataset output. The isinstance hashable is a fancier version of checking for string given `dataset[("var", "name")]` is valid given there is a variable named `("var", "name")` (tuples can be variable names too) but it is not valid if there are two variables, one "var" the other "name" in which case `ds[["var", "name"]]` is valid.

We might also want to make a helper function for that as it is quite common. I don't think it would fit in the validation module though, maybe we could have a scaffolding module too? cc @amaloney

Reply:
We certainly can create a scaffolding module.

I am confused as to why are we checking to see if `var_names` is hashable, and more generally, what types do we expect `var_names` to have? I think I've only ever encountered them being strings or tuples. Yes a tuple is hashable, but I think it would be easier to understand if we had a type for `var_names` so that when we checked

```python
if isinstance(var_names, VarNames): ...
```

it would make semantic sense. Then we can define `VarNames` using pydantic, or a dataclass, or some of the newer features in 3.12

Reply:
It is either of these 3: None (100% ours to indicate no filtering), `Hashable` -> select a single variable (inherited from xarray, and if you do `dataset[hashable]` you get a DataArray), `Sequence[Hashable]` -> select multiple variables (also from xarray, now if you do `dataset[sequence]` you get another dataset). 

If we forced users to interface directly with the xarray objects we wouldn't have to care about this, but the `var_names` and `coords` are two of the main arguments, especially `var_names` for which we allow excluding variables and fancier filtering which would be much harder to achieve with xarray. 

In general I think we want to maintain the xarray behaviour: None or sequence of hashable returns a dataset, hashable returns a dataarray. But there are cases like here where we do always want a dataset for internal use which won't be returned to the user. So we need to know what is valid in `dataset[xyz]` and what will be its result so we can adapt our code to these two cases as needed.